daml-lf/spec/daml-lf-1.rst

```diff
@@ -890,11 +890,13 @@ Then we define *well-formed expressions*. ::
     ——————————————————————————————————————————————————————————————— ExpOptionSome
       Γ  ⊢  'Some' @τ e  :  'Option' τ
 
-      ε ⊢ τ : *     Γ  ⊢  e  : τ
+      τ contains neither variables nor quantifiers
+      Γ  ⊢  τ : *     Γ  ⊢  e  : τ
     ——————————————————————————————————————————————————————————————— ExpToAny
       Γ  ⊢  'to_any' @τ e  :  'Any'
 
-      ε ⊢ τ : *     Γ  ⊢  e  : Any
+      τ contains neither variables nor quantifiers
+      Γ  ⊢  τ : *     Γ  ⊢  e  : Any
```

Comment:
We should probably get the old preconditions back and add `τ does not contain quantifiers`.

Reply:
I’ll change it together with changing the Haskell typechecker.

Reply:
(I forget the haskell typechecker. )

This is closer to what I implement in scala. 
Fell free to change everything as you which.